### PR TITLE
[MU4] Implemented standard dialogs

### PR DESCRIPTION
**Pull request by Eism**
https://user-images.githubusercontent.com/10116828/119487261-705fd600-bd59-11eb-844b-e0439bc3d3f9.mp4

### CODE REVIEW

File: src/framework/global/internal/interactive.cpp [hunk 1]
```diff
@@ -38,10 +38,21 @@
 using namespace mu;
 using namespace mu::framework;
 
+IInteractive::Result standardDialogResult(const RetVal<Val>& retVal)
```
Comment (igorkorsukov) [inline src/framework/global/internal/interactive.cpp:41]: Better to add static

File: src/framework/global/internal/interactive.cpp [hunk 2]
```diff
@@ -38,10 +38,21 @@
...
+{
+    if (!retVal.ret) {
+        return IInteractive::Result();
+    }
+
+    int btn = retVal.val.toQVariant().toMap()["buttonId"].toInt();
+    bool showAgain = retVal.val.toQVariant().toMap()["showAgain"].toBool();
```
Comment (igorkorsukov) [inline src/framework/global/internal/interactive.cpp:48]: better convert to map one time

File: src/framework/ui/qml/MuseScore/Ui/InteractiveProvider.qml [hunk 3]
```diff
@@ -59,40 +59,41 @@ Item {
...
-                var comp = Qt.createComponent("../../" + page.path);
-                if (comp.status !== Component.Ready) {
-                    console.log("[qml] failed create component: " + page.path + ", err: " + comp.errorString())
-                    data.setValue("ret", {errcode: 102 }) // CreateFailed
-                    return;
-                }
-
-                var obj = comp.createObject(root.topParent, page.params);
-                obj.objectID = root.provider.objectID(obj)
+                var dialogObj = createDialog(dialogPath, page.params)
```
Comment (igorkorsukov) [inline src/framework/ui/qml/MuseScore/Ui/InteractiveProvider.qml:64]: Formally, there can be more than just dialogues

File: src/framework/ui/view/interactiveprovider.cpp [hunk 4]
```diff
@@ -282,6 +356,36 @@ RetVal<InteractiveProvider::OpenData> InteractiveProvider::openQml(const UriQuer
...
 }
 
+RetVal<Val> InteractiveProvider::openStandardDialog(const QString& type, const QString& title, const framework::IInteractive::Text& text,
+                                                    const framework::IInteractive::ButtonDatas& buttons, int defBtn,
+                                                    const framework::IInteractive::Options& options)
+{
+    QmlLaunchData* data = new QmlLaunchData();
+    fillStandatdDialogData(data, type, title, text, buttons, defBtn, options);
+
+    emit fireOpenDialog(data);
```
Comment (igorkorsukov) [inline src/framework/ui/view/interactiveprovider.cpp:366]: I believe that it is actually better to just build uri and open it like any other dialogue.
Like:
```
    UriQuery q("musescore://interactive/error");
    q.addParam("title", title);
    q.addParam("text", text.toJson());
    q.addParam("buttons", buttons.toJson());
    q.addParam("defBtn", defBtn);
    q.addParam("options", options.toJson());

```

Comment (igorkorsukov) [reply src/framework/ui/view/interactiveprovider.cpp:366]: This is a desire, I do not insist :)

**Review by RomanPudashkin — APPROVED** [review]

**Pull request merged**